zebrad/src/components/sync.rs

```diff
@@ -16,38 +17,37 @@ use zebra_chain::{
 use zebra_network as zn;
 use zebra_state as zs;
 
+mod downloads;
+use downloads::{AlwaysHedge, Downloads};
+
 /// Controls the number of peers used for each ObtainTips and ExtendTips request.
 const FANOUT: usize = 4;
 
 /// Controls how many times we will retry each block download.
 ///
-/// If all the retries fail, then the syncer will reset, and start downloading
-/// blocks from the verified tip in the state, including blocks which previously
-/// downloaded successfully.
-///
-/// But if a node is on a slow or unreliable network, sync restarts can result
-/// in a flood of download requests, making future syncs more likely to fail.
-/// So it's much faster to retry each block multiple times.
+/// Failing block downloads is important because it defends against peers who
+/// feed us bad hashes. But spurious failures of valid blocks cause the syncer to
+/// restart from the previous checkpoint, potentially re-downloading blocks.
 ///
-/// When we implement a peer reputation system, we can reduce the number of
-/// retries, because we will be more likely to choose a good peer.
-const BLOCK_DOWNLOAD_RETRY_LIMIT: usize = 5;
+/// We also hedge requests, so we may retry up to twice this many times.
+const BLOCK_DOWNLOAD_RETRY_LIMIT: usize = 2;
```

Comment:
It would be nice to get rid of the outer Retry, but I think that would mean adding a retry limit within the Hedge.

And maybe we'd also want to set a mandatory number of retries, even when there are too few data points in the histogram.

Reply:
But that's nowhere near what Hedge can do now, so we should keep the outer Retry.

Reply:
I don't follow -- hedged requests send up to one additional request, what would a retry limit within the Hedge look like?

Reply:
Instead of having an outer retry and an inner hedge with 1 additional request, the hedge could send up to `N` additional requests.

Maybe that's actually more complicated though - because you have to decide when to send the other requests.

This is just speculation - it's not that important.